# Prerequisites {-}

The course is intended for those who have basic familiarity with Unix and 
the R scripting language.

* If a refresher is needed for Unix command line (hopefully not), please go over [this tutorial](https://ryanstutorials.net/linuxtutorial/) and its [companion cheatsheet](https://ryanstutorials.net/linuxtutorial/cheatsheet.php).
* Getting down to basics: an introduction to the fundamentals of R ([courtesy of Mark Ravinet](markravinet.github.io/Introduction.html)).
* Gentle introduction to `R/Biocondutor`: [here](https://bioconductor.github.io/BiocWorkshops/introduction-to-bioconductor-annotation-resources.html)
* For a full in-depth guide of `Bioconductor` ecosystem: read the comprehensive `R/Bioconductor` book from Kasper D. Hansen available under the CC BY-NC-SA 4.0 license. It is freely available under the [CC BY-NC-SA 4.0 license](https://creativecommons.org/licenses/by-nc-sa/4.0/deed.en_US): [Link to EPUB](https://leanpub.com/bioconductor)

## Local configuration 

* Computer with high-speed internet access 
* Zoom visioconference software

* **For Windows users: PLEASE INSTALL WSL 2**. Instructions are here: [learn.microsoft.com/en-us/windows/wsl/install](https://learn.microsoft.com/en-us/windows/wsl/install)

<!-- ::: {.callout-important icon='true'}
## Docker image
If trainees are familiar with `docker` and prefer using a pre-compiled image, see here. 
::: -->

* IGV software installed 

* Several system dependencies

    - `R>=4.1`: https://cloud.r-project.org/
    - `fastqc`: https://www.bioinformatics.babraham.ac.uk/projects/fastqc/
    - `cutadapt`: https://cutadapt.readthedocs.io/en/stable/installation.html
    - `trim_galore`: https://github.com/FelixKrueger/TrimGalore
    - `bowtie2`: https://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#obtaining-bowtie-2
    - `deepTools`: https://deeptools.readthedocs.io/en/develop/content/installation.html
    - `macs2`: https://github.com/macs3-project/MACS/wiki/Install-macs2#easy-installation-through-pypi
    - `meme`: https://meme-suite.org/meme/doc/install.html?man_type=web
    - `subread`: https://subread.sourceforge.net/

* Several R packages, the main ones being: 

    - `SummarizedExperiment`
    - `rtracklayer`
    - `GenomicRanges`
    - `GenomicAlignments`
    - `GenomicFeatures`
    - `DESeq2`
    - `gprofiler2`
