<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.433">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="J. Serizay">

<title>Epigenomics Data Analysis - 13&nbsp; Lab 3 - ChIP-seq downstream analysis</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../">
<link href="../../content/day4/Lecture8_RNAseq.html" rel="next">
<link href="../../content/day3/Demo3_ChIP-seq_processing.html" rel="prev">
<script src="../../site_libs/quarto-html/quarto.js"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


</head>

<body class="nav-sidebar floating">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
      <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../../content/day3/Lecture6_ChIPseq.html">Day 3</a></li><li class="breadcrumb-item"><a href="../../content/day3/Lab3_ChIP-seq_analysis.html"><span class="chapter-number">13</span>&nbsp; <span class="chapter-title">Lab 3 - ChIP-seq downstream analysis</span></a></li></ol></nav>
      <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
      </a>
      <button type="button" class="btn quarto-search-button" aria-label="" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="../../">Epigenomics Data Analysis</a> 
        <div class="sidebar-tools-main">
    <a href="https://github.com/js2264/NGS-Physalia/" rel="" title="Source Code" class="quarto-navigation-tool px-1" aria-label="Source Code"><i class="bi bi-github"></i></a>
</div>
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Welcome</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../content/program.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Program</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../content/Rstudio.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">RStudio</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../content/prerequisites.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Prerequisites</span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true">
 <span class="menu-text">Day 1</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../content/day1/Lecture1_introduction.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Lecture 1 - Introduction to epigenomics data analysis</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../content/day1/Lecture2_processing.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Lecture 2 - General processing of genome-wide epigenomic assays</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../content/day1/Demo1_MNase-seq_processing.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Demo 1 - Processing of MNase-seq data</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../content/day1/Lab1_MNase-seq_processing.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Lab 1 - Processing of MNase-seq data</span></span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true">
 <span class="menu-text">Day 2</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../content/day2/Lecture3_workflows.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">Lecture 3 - NGS processing workflows</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../content/day2/Lecture4_ATACseq.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">Lecture 4 - ATAC-seq processing and analyis</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../content/day2/Lecture5_R-101.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">Lecture 5 - R 101</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../content/day2/Demo2_ATAC-seq_processing.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">8</span>&nbsp; <span class="chapter-title">Demo 2 - Processing of ATAC-seq data</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../content/day2/Lab2_ATAC-seq_analysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">9</span>&nbsp; <span class="chapter-title">Lab 2 - ATAC-seq downstream analysis</span></span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="true">
 <span class="menu-text">Day 3</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../content/day3/Lecture6_ChIPseq.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">10</span>&nbsp; <span class="chapter-title">Lecture 6 - ChIP-seq processing and analyis</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../content/day3/Lecture7_R-201.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">11</span>&nbsp; <span class="chapter-title">Lecture 7 - R 201</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../content/day3/Demo3_ChIP-seq_processing.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">12</span>&nbsp; <span class="chapter-title">Demo 3 - Processing of ChIP-seq data</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../content/day3/Lab3_ChIP-seq_analysis.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text"><span class="chapter-number">13</span>&nbsp; <span class="chapter-title">Lab 3 - ChIP-seq downstream analysis</span></span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" aria-expanded="true">
 <span class="menu-text">Day 4</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-4" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../content/day4/Lecture8_RNAseq.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">14</span>&nbsp; <span class="chapter-title">Lecture 8 - RNA-seq processing and analyis</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../content/day4/Lecture9_R-301.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">15</span>&nbsp; <span class="chapter-title">Lecture 9 - R 301</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../content/day4/Demo4_RNA-seq_processing.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">16</span>&nbsp; <span class="chapter-title">Demo 4 - Processing of RNA-seq data</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../content/day4/Lab4_RNA-seq_analysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">17</span>&nbsp; <span class="chapter-title">Lab 4 - RNA-seq downstream analysis</span></span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <span class="sidebar-item-text sidebar-link text-start">
 <span class="menu-text">Day 5</span></span>
  </li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../content/extra.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Extra resources</span></a>
  </div>
</li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#download-peaks-from-encode-database" id="toc-download-peaks-from-encode-database" class="nav-link active" data-scroll-target="#download-peaks-from-encode-database"><span class="header-section-number">13.1</span> Download peaks from ENCODE database</a></li>
  <li><a href="#find-motifs-enriched-in-xnd-1-chip-seq" id="toc-find-motifs-enriched-in-xnd-1-chip-seq" class="nav-link" data-scroll-target="#find-motifs-enriched-in-xnd-1-chip-seq"><span class="header-section-number">13.2</span> Find motifs enriched in <code>xnd-1</code> ChIP-seq</a>
  <ul class="collapse">
  <li><a href="#preparing-meme-input" id="toc-preparing-meme-input" class="nav-link" data-scroll-target="#preparing-meme-input"><span class="header-section-number">13.2.1</span> Preparing <code>meme</code> input</a></li>
  <li><a href="#running-meme" id="toc-running-meme" class="nav-link" data-scroll-target="#running-meme"><span class="header-section-number">13.2.2</span> Running <code>meme</code></a></li>
  </ul></li>
  <li><a href="#compare-all-peaks-to-genomic-features" id="toc-compare-all-peaks-to-genomic-features" class="nav-link" data-scroll-target="#compare-all-peaks-to-genomic-features"><span class="header-section-number">13.3</span> Compare all peaks to genomic features</a>
  <ul class="collapse">
  <li><a href="#import-all-peaks-in-r" id="toc-import-all-peaks-in-r" class="nav-link" data-scroll-target="#import-all-peaks-in-r"><span class="header-section-number">13.3.1</span> Import all peaks in R</a></li>
  <li><a href="#define-genomic-features-in-ce11" id="toc-define-genomic-features-in-ce11" class="nav-link" data-scroll-target="#define-genomic-features-in-ce11"><span class="header-section-number">13.3.2</span> Define genomic features in <code>ce11</code></a></li>
  <li><a href="#chip-seq-peak-overlaps-with-genomic-features" id="toc-chip-seq-peak-overlaps-with-genomic-features" class="nav-link" data-scroll-target="#chip-seq-peak-overlaps-with-genomic-features"><span class="header-section-number">13.3.3</span> ChIP-seq peak overlaps with genomic features</a></li>
  <li><a href="#peaks-occurrence-over-tissue-specific-regulatory-elements" id="toc-peaks-occurrence-over-tissue-specific-regulatory-elements" class="nav-link" data-scroll-target="#peaks-occurrence-over-tissue-specific-regulatory-elements"><span class="header-section-number">13.3.4</span> Peaks occurrence over tissue-specific regulatory elements</a></li>
  </ul></li>
  </ul>
<div class="toc-actions"><div><i class="bi bi-github"></i></div><div class="action-links"><p><a href="https://github.com/js2264/NGS-Physalia/edit/main/content/day3/Lab3_ChIP-seq_analysis.qmd" class="toc-action">Edit this page</a></p><p><a href="https://github.com/js2264/NGS-Physalia/issues/new" class="toc-action">Report an issue</a></p></div></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title"><span class="chapter-number">13</span>&nbsp; <span class="chapter-title">Lab 3 - ChIP-seq downstream analysis</span></h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>

<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Aims
</div>
</div>
<div class="callout-body-container callout-body">
<ul>
<li>Find motifs enriched in a set of ChIP-seq peaks</li>
<li>Import a dozen of ChIP-seq peak sets in R</li>
<li>Check distribution of peaks comapred to genomic features</li>
<li>Check peak occurrence over tissue-specific regulatory elements</li>
</ul>
</div>
</div>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Datasets
</div>
</div>
<div class="callout-body-container callout-body">
<ul>
<li>modENCODE/modERN TF ChIP-seq database, available <a href="https://www.encodeproject.org/search/?type=Experiment&amp;control_type!=*&amp;status=released&amp;replicates.library.biosample.donor.organism.scientific_name=Caenorhabditis+elegans&amp;biosample_ontology.term_name=whole+organism&amp;assay_title=TF+ChIP-seq&amp;biosample_ontology.classification=whole%20organisms">here</a></li>
</ul>
</div>
</div>
<section id="download-peaks-from-encode-database" class="level2" data-number="13.1">
<h2 data-number="13.1" class="anchored" data-anchor-id="download-peaks-from-encode-database"><span class="header-section-number">13.1</span> Download peaks from ENCODE database</h2>
<p>Check out the <a href="https://www.encodeproject.org/search/?type=Experiment&amp;control_type!=*&amp;status=released&amp;replicates.library.biosample.donor.organism.scientific_name=Caenorhabditis+elegans&amp;biosample_ontology.term_name=whole+organism&amp;assay_title=TF+ChIP-seq&amp;biosample_ontology.classification=whole%20organisms">modENCODE/modERN TF ChIP-seq database</a>.</p>
<p>Inspect the available datasets, by filtering e.g.&nbsp;using the following criteria:</p>
<ul>
<li>Assay title: <code>TF ChIP-seq</code></li>
<li>Organism: <code>C. elegans</code></li>
<li>Genome assembly: <code>ce11</code></li>
<li>Project: <code>modENCODE</code></li>
</ul>
<p>Peak files for 12 ChIP-seq datasets have been already downloaded from this database.</p>
</section>
<section id="find-motifs-enriched-in-xnd-1-chip-seq" class="level2" data-number="13.2">
<h2 data-number="13.2" class="anchored" data-anchor-id="find-motifs-enriched-in-xnd-1-chip-seq"><span class="header-section-number">13.2</span> Find motifs enriched in <code>xnd-1</code> ChIP-seq</h2>
<ul>
<li>Check the <a href="https://meme-suite.org/"><code>meme</code> website</a> to identify which tool is best suited to identify a motif <em>de novo</em> in a set of peaks from a ChIP-seq experiment.</li>
<li>What do you need to run <code>xstreme</code> on a set of peaks?</li>
</ul>
<section id="preparing-meme-input" class="level3" data-number="13.2.1">
<h3 data-number="13.2.1" class="anchored" data-anchor-id="preparing-meme-input"><span class="header-section-number">13.2.1</span> Preparing <code>meme</code> input</h3>
<ul>
<li>Import <code>xnd-1</code> peaks in R as a <code>GRanges</code> object.</li>
<li>Recover <code>ce11</code> genome sequence using the <code>BSgenome.Celegans.UCSC.ce11</code> package.</li>
<li>Extract sequence over <code>xnd-1</code> peaks with the <code>Biostrings</code> package.</li>
<li>Export the sequences as a <code>fasta</code> file.</li>
</ul>
<div class="icon callout callout-style-default callout-answer no-icon callout-titled">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-3-contents" aria-controls="callout-3" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Answer
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-3" class="callout-3-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<div class="cell">
<div class="code-with-filename">
<div class="code-with-filename-file">
<pre><strong>r</strong></pre>
</div>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1"></a><span class="fu">library</span>(rtracklayer)</span>
<span id="cb1-2"><a href="#cb1-2"></a><span class="fu">library</span>(Biostrings)</span>
<span id="cb1-3"><a href="#cb1-3"></a><span class="fu">library</span>(BSgenome.Celegans.UCSC.ce11)</span>
<span id="cb1-4"><a href="#cb1-4"></a>xnd1 <span class="ot">&lt;-</span> <span class="fu">import</span>(<span class="st">'data/peaks/modENCODE/xnd-1.bed'</span>)</span>
<span id="cb1-5"><a href="#cb1-5"></a>genome <span class="ot">&lt;-</span> BSgenome.Celegans.UCSC.ce11</span>
<span id="cb1-6"><a href="#cb1-6"></a>xnd1_seq <span class="ot">&lt;-</span> <span class="fu">getSeq</span>(genome, xnd1)</span>
<span id="cb1-7"><a href="#cb1-7"></a><span class="fu">names</span>(xnd1_seq) <span class="ot">&lt;-</span> <span class="fu">as.character</span>(xnd1)</span>
<span id="cb1-8"><a href="#cb1-8"></a><span class="fu">writeXStringSet</span>(xnd1_seq, <span class="st">'data/peaks/modENCODE/xnd-1.fa'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</div>
</div>
</div>
</div>
</section>
<section id="running-meme" class="level3" data-number="13.2.2">
<h3 data-number="13.2.2" class="anchored" data-anchor-id="running-meme"><span class="header-section-number">13.2.2</span> Running <code>meme</code></h3>
<ul>
<li><p>Identify motifs enriched in <code>xnd-1</code> peaks with <code>xstreme</code> using the following options:</p>
<ul>
<li>Zero or one occurence per sequence at most for <code>meme</code></li>
<li>At most 3 motifs</li>
<li>A min motif width of 6 for <code>meme</code></li>
<li>A max motif width of 15 for <code>meme</code></li>
<li>No <code>streme</code> motifs</li>
<li>Over multiple processors in parallel</li>
</ul></li>
</ul>
<div class="icon callout callout-style-default callout-answer no-icon callout-titled">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-4-contents" aria-controls="callout-4" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Answer
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-4" class="callout-4-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<div class="cell">
<div class="code-with-filename">
<div class="code-with-filename-file">
<pre><strong>sh</strong></pre>
</div>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode numberSource bash number-lines code-with-copy"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1"></a><span class="ex">xstreme</span> <span class="dt">\</span></span>
<span id="cb2-2"><a href="#cb2-2"></a>    <span class="at">--p</span> data/peaks/modENCODE/xnd-1.fa <span class="dt">\</span></span>
<span id="cb2-3"><a href="#cb2-3"></a>    <span class="at">--o</span> data/meme/ <span class="dt">\</span></span>
<span id="cb2-4"><a href="#cb2-4"></a>    <span class="at">-minw</span> 12 <span class="dt">\</span></span>
<span id="cb2-5"><a href="#cb2-5"></a>    <span class="at">-maxw</span> 16 <span class="dt">\</span></span>
<span id="cb2-6"><a href="#cb2-6"></a>    <span class="at">--meme-nmotifs</span> 3 <span class="dt">\</span></span>
<span id="cb2-7"><a href="#cb2-7"></a>    <span class="at">--meme-p</span> 12 <span class="dt">\</span></span>
<span id="cb2-8"><a href="#cb2-8"></a>    <span class="at">--meme-mod</span> zoops <span class="dt">\</span></span>
<span id="cb2-9"><a href="#cb2-9"></a>    <span class="at">--streme-nmotifs</span> 0</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</div>
</div>
</div>
</div>
<ul>
<li>Check the results. Do they make sense in the light of recent publications? (see <a href="https://www.science.org/doi/10.1126/sciadv.abo4082">here</a>)</li>
</ul>
</section>
</section>
<section id="compare-all-peaks-to-genomic-features" class="level2" data-number="13.3">
<h2 data-number="13.3" class="anchored" data-anchor-id="compare-all-peaks-to-genomic-features"><span class="header-section-number">13.3</span> Compare all peaks to genomic features</h2>
<section id="import-all-peaks-in-r" class="level3" data-number="13.3.1">
<h3 data-number="13.3.1" class="anchored" data-anchor-id="import-all-peaks-in-r"><span class="header-section-number">13.3.1</span> Import all peaks in R</h3>
<ul>
<li>In R, list all the <code>bed</code> files available for the peaks.</li>
<li>Import each file as a <code>GRanges</code> object.</li>
</ul>
<div class="icon callout callout-style-default callout-answer no-icon callout-titled">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-5-contents" aria-controls="callout-5" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Answer
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-5" class="callout-5-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<div class="cell">
<div class="code-with-filename">
<div class="code-with-filename-file">
<pre><strong>r</strong></pre>
</div>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1"></a><span class="fu">library</span>(rtracklayer)</span>
<span id="cb3-2"><a href="#cb3-2"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb3-3"><a href="#cb3-3"></a>bedfiles <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="st">'data/peaks/modENCODE/'</span>, <span class="at">pattern =</span> <span class="st">'*bed'</span>)</span>
<span id="cb3-4"><a href="#cb3-4"></a>peaks <span class="ot">&lt;-</span> <span class="fu">map</span>(bedfiles, <span class="sc">~</span> <span class="fu">import</span>(<span class="fu">file.path</span>(<span class="st">'data/peaks/modENCODE/'</span>, .x)))</span>
<span id="cb3-5"><a href="#cb3-5"></a><span class="fu">names</span>(peaks) <span class="ot">&lt;-</span> <span class="fu">str_replace</span>(bedfiles, <span class="st">'.bed'</span>, <span class="st">''</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</div>
</div>
</div>
</div>
</section>
<section id="define-genomic-features-in-ce11" class="level3" data-number="13.3.2">
<h3 data-number="13.3.2" class="anchored" data-anchor-id="define-genomic-features-in-ce11"><span class="header-section-number">13.3.2</span> Define genomic features in <code>ce11</code></h3>
<p>Genomic features can be easily annotated from a set of gene features. <code>ChIPseeker</code> facilitates the annotation of ChIP-seq peaks using gene annotations directly provided in R by <code>TxDb</code> gene annotation packages, e.g.&nbsp;<code>TxDb.Celegans.UCSC.ce11.ensGene</code></p>
<ul>
<li>Install the <code>TxDb.Celegans.UCSC.ce11.ensGene</code> package in R</li>
<li>Install <code>ChIPseeker</code></li>
</ul>
<div class="icon callout callout-style-default callout-answer no-icon callout-titled">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-6-contents" aria-controls="callout-6" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Answer
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-6" class="callout-6-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<div class="cell">
<div class="code-with-filename">
<div class="code-with-filename-file">
<pre><strong>r</strong></pre>
</div>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">'TxDb.Celegans.UCSC.ce11.ensGene'</span>)</span>
<span id="cb4-2"><a href="#cb4-2"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">'ChIPseeker'</span>)</span>
<span id="cb4-3"><a href="#cb4-3"></a><span class="fu">library</span>(ChIPseeker)</span>
<span id="cb4-4"><a href="#cb4-4"></a><span class="fu">library</span>(TxDb.Celegans.UCSC.ce11.ensGene)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</div>
</div>
</div>
</div>
</section>
<section id="chip-seq-peak-overlaps-with-genomic-features" class="level3" data-number="13.3.3">
<h3 data-number="13.3.3" class="anchored" data-anchor-id="chip-seq-peak-overlaps-with-genomic-features"><span class="header-section-number">13.3.3</span> ChIP-seq peak overlaps with genomic features</h3>
<ul>
<li>Use <code>ChIPseeker</code> to annotate ChIP-seq peaks for a single ChIP-seq experiment.</li>
<li>Extract</li>
</ul>
<div class="icon callout callout-style-default callout-answer no-icon callout-titled">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-7-contents" aria-controls="callout-7" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Answer
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-7" class="callout-7-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<div class="cell">
<div class="code-with-filename">
<div class="code-with-filename-file">
<pre><strong>r</strong></pre>
</div>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1"></a>ceh48 <span class="ot">&lt;-</span> peaks[[<span class="dv">1</span>]]</span>
<span id="cb5-2"><a href="#cb5-2"></a>annots_ceh48 <span class="ot">&lt;-</span> <span class="fu">annotatePeak</span>(ceh48, <span class="at">TxDb =</span> TxDb.Celegans.UCSC.ce11.ensGene)</span>
<span id="cb5-3"><a href="#cb5-3"></a><span class="fu">getAnnoStat</span>(annots_ceh48)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</div>
</div>
</div>
</div>
<ul>
<li>Iterate over all the peak sets to compile their annotations across <em>C. elegans</em> genome. To iterate over each set of peaks in the <code>peaks</code> list and return an aggregate <code>data.frame</code>, use the <code>imap_dfr</code>.</li>
</ul>
<div class="icon callout callout-style-default callout-answer no-icon callout-titled">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-8-contents" aria-controls="callout-8" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Answer
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-8" class="callout-8-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<div class="cell">
<div class="code-with-filename">
<div class="code-with-filename-file">
<pre><strong>r</strong></pre>
</div>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1"></a>annots <span class="ot">&lt;-</span> <span class="fu">imap_dfr</span>(peaks, <span class="sc">~</span> <span class="fu">annotatePeak</span>(.x, <span class="at">TxDb =</span> TxDb.Celegans.UCSC.ce11.ensGene) <span class="sc">|&gt;</span> </span>
<span id="cb6-2"><a href="#cb6-2"></a>    <span class="fu">getAnnoStat</span>() <span class="sc">|&gt;</span> </span>
<span id="cb6-3"><a href="#cb6-3"></a>    <span class="fu">mutate</span>(<span class="at">TF =</span> .y)</span>
<span id="cb6-4"><a href="#cb6-4"></a>)</span>
<span id="cb6-5"><a href="#cb6-5"></a>annots</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</div>
</div>
</div>
</div>
<ul>
<li>Generate a barplot to represent the % of peaks in each type of genomic features, for the 12 TFs.</li>
</ul>
<div class="icon callout callout-style-default callout-answer no-icon callout-titled">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-9-contents" aria-controls="callout-9" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Answer
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-9" class="callout-9-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<div class="cell">
<div class="code-with-filename">
<div class="code-with-filename-file">
<pre><strong>r</strong></pre>
</div>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1"></a><span class="fu">ggplot</span>(annots, <span class="fu">aes</span>(<span class="at">x =</span> TF, <span class="at">y =</span> Frequency, <span class="at">fill =</span> Feature)) <span class="sc">+</span> </span>
<span id="cb7-2"><a href="#cb7-2"></a>    <span class="fu">geom_col</span>() <span class="sc">+</span> </span>
<span id="cb7-3"><a href="#cb7-3"></a>    <span class="fu">guides</span>(<span class="at">x =</span>  <span class="fu">guide_axis</span>(<span class="at">angle =</span> <span class="dv">90</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</div>
</div>
</div>
</div>
<ul>
<li>Comment on the distribution of the 12 sets of peaks over genomic features.</li>
</ul>
</section>
<section id="peaks-occurrence-over-tissue-specific-regulatory-elements" class="level3" data-number="13.3.4">
<h3 data-number="13.3.4" class="anchored" data-anchor-id="peaks-occurrence-over-tissue-specific-regulatory-elements"><span class="header-section-number">13.3.4</span> Peaks occurrence over tissue-specific regulatory elements</h3>
<ul>
<li>Import regulatory elements annotated in <em>C. elegans</em> in R, as seen in previous Lab.</li>
</ul>
<div class="icon callout callout-style-default callout-answer no-icon callout-titled">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-10-contents" aria-controls="callout-10" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Answer
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-10" class="callout-10-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<div class="cell">
<div class="code-with-filename">
<div class="code-with-filename-file">
<pre><strong>R</strong></pre>
</div>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1"></a><span class="fu">library</span>(readxl)</span>
<span id="cb8-2"><a href="#cb8-2"></a><span class="fu">download.file</span>(<span class="st">'https://genome.cshlp.org/content/suppl/2020/11/16/gr.265934.120.DC1/Supplemental_Table_S2.xlsx'</span>, <span class="st">'data/WBcel235_REs.xlsx'</span>)</span>
<span id="cb8-3"><a href="#cb8-3"></a>REs <span class="ot">&lt;-</span> <span class="fu">read_xlsx</span>(<span class="st">'data/WBcel235_REs.xlsx'</span>, <span class="at">skip =</span> <span class="dv">2</span>, <span class="at">col_names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb8-4"><a href="#cb8-4"></a>REs <span class="ot">&lt;-</span> <span class="fu">makeGRangesFromDataFrame</span>(</span>
<span id="cb8-5"><a href="#cb8-5"></a>    REs, <span class="at">seqnames.field =</span> <span class="st">'chrom_ce11'</span>, </span>
<span id="cb8-6"><a href="#cb8-6"></a>    <span class="at">start.field =</span> <span class="st">'start_ce11'</span>, <span class="at">end.field =</span> <span class="st">'end_ce11'</span>,</span>
<span id="cb8-7"><a href="#cb8-7"></a>    <span class="at">keep.extra.columns =</span> <span class="cn">TRUE</span></span>
<span id="cb8-8"><a href="#cb8-8"></a>)</span>
<span id="cb8-9"><a href="#cb8-9"></a>REs</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</div>
</div>
</div>
</div>
<ul>
<li>Check for overlap of each regulatory element with <code>xnd-1</code> peaks.</li>
<li>Check the overlap of germline-specific regulatory element with <code>xnd-1</code> peaks.</li>
<li>Check the <strong>enrichment</strong> of germline-specific regulatory element over <code>xnd-1</code> peaks.</li>
</ul>
<div class="icon callout callout-style-default callout-answer no-icon callout-titled">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-11-contents" aria-controls="callout-11" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Answer
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-11" class="callout-11-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<div class="cell">
<div class="code-with-filename">
<div class="code-with-filename-file">
<pre><strong>R</strong></pre>
</div>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1"></a><span class="fu">table</span>(REs <span class="sc">%over%</span> peaks[[<span class="st">'xnd-1'</span>]])</span>
<span id="cb9-2"><a href="#cb9-2"></a><span class="fu">table</span>(REs <span class="sc">%over%</span> peaks[[<span class="st">'xnd-1'</span>]], REs<span class="sc">$</span>Annotation)</span>
<span id="cb9-3"><a href="#cb9-3"></a><span class="fu">table</span>(REs <span class="sc">%over%</span> peaks[[<span class="st">'xnd-1'</span>]], REs<span class="sc">$</span>Annotation <span class="sc">==</span> <span class="st">'Germline'</span>) </span>
<span id="cb9-4"><a href="#cb9-4"></a><span class="fu">table</span>(REs <span class="sc">%over%</span> peaks[[<span class="st">'xnd-1'</span>]], REs<span class="sc">$</span>Annotation <span class="sc">==</span> <span class="st">'Germline'</span>) <span class="sc">|&gt;</span> <span class="fu">fisher.test</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</div>
</div>
</div>
</div>
<ul>
<li>For each tissue (<code>Germline</code>, <code>Neurons</code>, <code>Muscle</code>, <code>Hypod.</code> and <code>Intest.</code>), check whether the tissue-specific REs are enriched in <code>xnd-1</code> peak.</li>
</ul>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Tip
</div>
</div>
<div class="callout-body-container callout-body">
<ul>
<li>You can iterate over each <code>tissue</code> and return an aggregated <code>data.frame</code> using <code>map_dfr</code>.</li>
<li>For each iteration, you can transform the result of <code>fisher.test()</code> into a <code>tibble</code> with the <code>glance</code> function from <code>broom</code> package.</li>
</ul>
</div>
</div>
<div class="icon callout callout-style-default callout-answer no-icon callout-titled">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-13-contents" aria-controls="callout-13" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Answer
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-13" class="callout-13-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<div class="cell">
<div class="code-with-filename">
<div class="code-with-filename-file">
<pre><strong>R</strong></pre>
</div>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1"></a>tissues <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Germline"</span>, <span class="st">"Neurons"</span>, <span class="st">"Muscle"</span>, <span class="st">"Hypod."</span>, <span class="st">"Intest."</span>)</span>
<span id="cb10-2"><a href="#cb10-2"></a>df <span class="ot">&lt;-</span> <span class="fu">map_dfr</span>(tissues, <span class="cf">function</span>(tissue) {</span>
<span id="cb10-3"><a href="#cb10-3"></a>    <span class="fu">fisher.test</span>(</span>
<span id="cb10-4"><a href="#cb10-4"></a>        REs <span class="sc">%over%</span> peaks[[<span class="st">'xnd-1'</span>]], </span>
<span id="cb10-5"><a href="#cb10-5"></a>        REs<span class="sc">$</span>Annotation <span class="sc">==</span> tissue</span>
<span id="cb10-6"><a href="#cb10-6"></a>    ) <span class="sc">|&gt;</span> </span>
<span id="cb10-7"><a href="#cb10-7"></a>        broom<span class="sc">::</span><span class="fu">glance</span>() <span class="sc">|&gt;</span> </span>
<span id="cb10-8"><a href="#cb10-8"></a>        <span class="fu">mutate</span>(<span class="at">tissue =</span> tissue, <span class="at">TF =</span> <span class="st">'xnd-1'</span>) <span class="sc">|&gt;</span> </span>
<span id="cb10-9"><a href="#cb10-9"></a>        dplyr<span class="sc">::</span><span class="fu">select</span>(tissue, TF, estimate, p.value)</span>
<span id="cb10-10"><a href="#cb10-10"></a>})</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</div>
</div>
</div>
</div>
<ul>
<li>Perform the same operation by iterating over each peak set in the list of imported peaks.</li>
</ul>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Tip
</div>
</div>
<div class="callout-body-container callout-body">
<ul>
<li>You can use two nested <code>map_dfr</code>, iterating first over each TF then over each tissue.</li>
</ul>
</div>
</div>
<div class="icon callout callout-style-default callout-answer no-icon callout-titled">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-15-contents" aria-controls="callout-15" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Answer
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-15" class="callout-15-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<div class="cell">
<div class="code-with-filename">
<div class="code-with-filename-file">
<pre><strong>R</strong></pre>
</div>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1"></a>TFs <span class="ot">&lt;-</span> <span class="fu">names</span>(peaks)</span>
<span id="cb11-2"><a href="#cb11-2"></a>tissues <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Germline"</span>, <span class="st">"Neurons"</span>, <span class="st">"Muscle"</span>, <span class="st">"Hypod."</span>, <span class="st">"Intest."</span>)</span>
<span id="cb11-3"><a href="#cb11-3"></a>df <span class="ot">&lt;-</span> <span class="fu">map_dfr</span>(TFs, <span class="cf">function</span>(TF) {</span>
<span id="cb11-4"><a href="#cb11-4"></a>    <span class="fu">map_dfr</span>(tissues, <span class="cf">function</span>(tissue) {</span>
<span id="cb11-5"><a href="#cb11-5"></a>        <span class="fu">fisher.test</span>(</span>
<span id="cb11-6"><a href="#cb11-6"></a>            REs <span class="sc">%over%</span> peaks[[TF]], </span>
<span id="cb11-7"><a href="#cb11-7"></a>            REs<span class="sc">$</span>Annotation <span class="sc">==</span> tissue</span>
<span id="cb11-8"><a href="#cb11-8"></a>        ) <span class="sc">|&gt;</span> </span>
<span id="cb11-9"><a href="#cb11-9"></a>            broom<span class="sc">::</span><span class="fu">glance</span>() <span class="sc">|&gt;</span> </span>
<span id="cb11-10"><a href="#cb11-10"></a>            <span class="fu">mutate</span>(<span class="at">tissue =</span> tissue, <span class="at">TF =</span> TF) <span class="sc">|&gt;</span> </span>
<span id="cb11-11"><a href="#cb11-11"></a>            dplyr<span class="sc">::</span><span class="fu">select</span>(tissue, TF, estimate, p.value)</span>
<span id="cb11-12"><a href="#cb11-12"></a>    })</span>
<span id="cb11-13"><a href="#cb11-13"></a>})</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</div>
</div>
</div>
</div>
<ul>
<li>For each TF, filter the tissues in which it is preferentially enriched (odds ratio &gt;= 2, p.value &lt;= 0.05)</li>
<li>Find TFs enriched over Intestine REs</li>
</ul>
<div class="icon callout callout-style-default callout-answer no-icon callout-titled">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-16-contents" aria-controls="callout-16" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Answer
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-16" class="callout-16-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<div class="cell">
<div class="code-with-filename">
<div class="code-with-filename-file">
<pre><strong>R</strong></pre>
</div>
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1"></a><span class="fu">filter</span>(df, estimate <span class="sc">&gt;=</span> <span class="dv">2</span>, p.value <span class="sc">&lt;=</span> <span class="fl">0.05</span>)</span>
<span id="cb12-2"><a href="#cb12-2"></a><span class="fu">filter</span>(df, estimate <span class="sc">&gt;=</span> <span class="dv">2</span>, p.value <span class="sc">&lt;=</span> <span class="fl">0.05</span>, tissue <span class="sc">==</span> <span class="st">"Intest."</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</div>
</div>
</div>
</div>
<ul>
<li>Check the <a href="https://string-db.org/cgi/input?input_page_active_form=multiple_identifiers">STRING DB website</a> to assess whether these TFs have been shown to interact together. Comment.</li>
</ul>


</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="../../content/day3/Demo3_ChIP-seq_processing.html" class="pagination-link">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">12</span>&nbsp; <span class="chapter-title">Demo 3 - Processing of ChIP-seq data</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="../../content/day4/Lecture8_RNAseq.html" class="pagination-link">
        <span class="nav-page-text"><span class="chapter-number">14</span>&nbsp; <span class="chapter-title">Lecture 8 - RNA-seq processing and analyis</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-left">Epigenomics Data Analysis |<br>
J. Serizay</div>   
    <div class="nav-footer-center">
      &nbsp;
    </div>
    <div class="nav-footer-right">This book was built with <a href="https://quarto.org/">Quarto</a>.</div>
  </div>
</footer>



</body></html>