<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.433">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="J. Serizay">

<title>Epigenomics Data Analysis - 21&nbsp; Lab 5 - Multi-omics data integration</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../">
<link href="../../content/extra.html" rel="next">
<link href="../../content/day5/Demo5_Bioc-resources.html" rel="prev">
<script src="../../site_libs/quarto-html/quarto.js"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


</head>

<body class="nav-sidebar floating">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
      <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../../content/day5/Lecture10_HiC.html">Day 5</a></li><li class="breadcrumb-item"><a href="../../content/day5/Lab5_data-integration.html"><span class="chapter-number">21</span>&nbsp; <span class="chapter-title">Lab 5 - Multi-omics data integration</span></a></li></ol></nav>
      <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
      </a>
      <button type="button" class="btn quarto-search-button" aria-label="" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="../../">Epigenomics Data Analysis</a> 
        <div class="sidebar-tools-main">
    <a href="https://github.com/js2264/NGS-Physalia/" rel="" title="Source Code" class="quarto-navigation-tool px-1" aria-label="Source Code"><i class="bi bi-github"></i></a>
</div>
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Welcome</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../content/program.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Program</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../content/Rstudio.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">RStudio</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../content/prerequisites.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Prerequisites</span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true">
 <span class="menu-text">Day 1</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../content/day1/Lecture1_introduction.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Lecture 1 - Introduction to epigenomics data analysis</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../content/day1/Lecture2_processing.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Lecture 2 - General processing of genome-wide epigenomic assays</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../content/day1/Demo1_MNase-seq_processing.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Demo 1 - Processing of MNase-seq data</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../content/day1/Lab1_MNase-seq_processing.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Lab 1 - Processing of MNase-seq data</span></span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true">
 <span class="menu-text">Day 2</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../content/day2/Lecture3_workflows.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">Lecture 3 - NGS processing workflows</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../content/day2/Lecture4_ATACseq.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">Lecture 4 - ATAC-seq processing and analyis</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../content/day2/Lecture5_R-101.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">Lecture 5 - R 101</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../content/day2/Demo2_ATAC-seq_processing.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">8</span>&nbsp; <span class="chapter-title">Demo 2 - Processing of ATAC-seq data</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../content/day2/Lab2_ATAC-seq_analysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">9</span>&nbsp; <span class="chapter-title">Lab 2 - ATAC-seq downstream analysis</span></span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="true">
 <span class="menu-text">Day 3</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../content/day3/Lecture6_ChIPseq.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">10</span>&nbsp; <span class="chapter-title">Lecture 6 - ChIP-seq processing and analyis</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../content/day3/Lecture7_R-201.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">11</span>&nbsp; <span class="chapter-title">Lecture 7 - R 201</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../content/day3/Demo3_ChIP-seq_processing.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">12</span>&nbsp; <span class="chapter-title">Demo 3 - Processing of ChIP-seq data</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../content/day3/Lab3_ChIP-seq_analysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">13</span>&nbsp; <span class="chapter-title">Lab 3 - ChIP-seq downstream analysis</span></span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" aria-expanded="true">
 <span class="menu-text">Day 4</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-4" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../content/day4/Lecture8_RNAseq.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">14</span>&nbsp; <span class="chapter-title">Lecture 8 - RNA-seq processing and analyis</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../content/day4/Lecture9_R-301.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">15</span>&nbsp; <span class="chapter-title">Lecture 9 - R 301</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../content/day4/Demo4_RNA-seq_processing.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">16</span>&nbsp; <span class="chapter-title">Demo 4 - Processing of RNA-seq data</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../content/day4/Lab4_RNA-seq_analysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">17</span>&nbsp; <span class="chapter-title">Lab 4 - RNA-seq downstream analysis</span></span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" aria-expanded="true">
 <span class="menu-text">Day 5</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-5" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../content/day5/Lecture10_HiC.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">18</span>&nbsp; <span class="chapter-title">Lecture 10 - HiC processing</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../content/day5/Lecture11_GO-GSEA.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">19</span>&nbsp; <span class="chapter-title">Lecture 11 - GO and GSEA analysis</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../content/day5/Demo5_Bioc-resources.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">20</span>&nbsp; <span class="chapter-title">Demo 5 - Leveraging Bioconductor resources</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../content/day5/Lab5_data-integration.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text"><span class="chapter-number">21</span>&nbsp; <span class="chapter-title">Lab 5 - Multi-omics data integration</span></span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../content/extra.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Extra resources</span></a>
  </div>
</li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#inspecting-data-in-igv" id="toc-inspecting-data-in-igv" class="nav-link active" data-scroll-target="#inspecting-data-in-igv"><span class="header-section-number">21.1</span> Inspecting data in IGV</a></li>
  <li><a href="#plotting-stranded-rna-seq-signal-over-scc1-chip-seq-peaks" id="toc-plotting-stranded-rna-seq-signal-over-scc1-chip-seq-peaks" class="nav-link" data-scroll-target="#plotting-stranded-rna-seq-signal-over-scc1-chip-seq-peaks"><span class="header-section-number">21.2</span> Plotting stranded RNA-seq signal over Scc1 ChIP-seq peaks</a></li>
  <li><a href="#plot-a-heaetmap-of-rna-seq-coverage-over-ordered-scc1-peaks" id="toc-plot-a-heaetmap-of-rna-seq-coverage-over-ordered-scc1-peaks" class="nav-link" data-scroll-target="#plot-a-heaetmap-of-rna-seq-coverage-over-ordered-scc1-peaks"><span class="header-section-number">21.3</span> Plot a heaetmap of RNA-seq coverage over ordered Scc1 peaks</a></li>
  <li><a href="#plot-stranded-rna-seq-coverages-for-scc1-peaks-grouped-by-their-strength" id="toc-plot-stranded-rna-seq-coverages-for-scc1-peaks-grouped-by-their-strength" class="nav-link" data-scroll-target="#plot-stranded-rna-seq-coverages-for-scc1-peaks-grouped-by-their-strength"><span class="header-section-number">21.4</span> Plot stranded RNA-seq coverages for Scc1 peaks grouped by their strength</a></li>
  </ul>
<div class="toc-actions"><div><i class="bi bi-github"></i></div><div class="action-links"><p><a href="https://github.com/js2264/NGS-Physalia/edit/main/content/day5/Lab5_data-integration.qmd" class="toc-action">Edit this page</a></p><p><a href="https://github.com/js2264/NGS-Physalia/issues/new" class="toc-action">Report an issue</a></p></div></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title"><span class="chapter-number">21</span>&nbsp; <span class="chapter-title">Lab 5 - Multi-omics data integration</span></h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>

<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Aims
</div>
</div>
<div class="callout-body-container callout-body">
<ul>
<li>Visually inspect results from MNase-seq, Scc1 ChIP-seq and RNA-seq in yeast</li>
<li>Plot aggregated profiles of stranded RNA-seq coverage @ Scc1 ChIP-seq peaks</li>
<li>Compare stranded RNA-seq coverages at strong/weak Scc1 peaks</li>
</ul>
</div>
</div>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Datasets
</div>
</div>
<div class="callout-body-container callout-body">
<ul>
<li>Scc1 ChIP-seq: unpublished</li>
<li>MNase-seq: unpublished</li>
<li>RNA-seq from <a href="https://www.nature.com/articles/s41597-020-0412-z">Nuño-Cabanes et al., Scientific Data 2020</a>: <code>SRR9929263</code></li>
</ul>
</div>
</div>
<!-- 
- Scc1 track: /data/20220309_Christophe_GC-paper/data/S288c_Mpneumo/ChIP/tracks/CH219/CH219%5Eunmapped_CBS138%5Emapped_S288c_Mpneumo%5EEHEJFT.CPM.bw
- Scc1 peaks: /data/20220309_Christophe_GC-paper/data/S288c_Mpneumo/ChIP/peaks/CH219/CH219_vs-CH220_genome-S288c_Mpneumo_EHEJFT_peaks.narrowPeak
- MNase track: /data/20220309_Christophe_GC-paper/data/S288c_Mpneumo/MNase/tracks/CH206/CH206%5Emapped_S288c_Mpneumo%5Efiltered%5E130-200%5EHS5L1A.130-200.CPM.bw

cp /data/20220309_Christophe_GC-paper/data/S288c_Mpneumo/ChIP/tracks/CH219/CH219^unmapped_CBS138^mapped_S288c_Mpneumo^EHEJFT.CPM.bw EpigenomicsDataAnalysis_2023/data/day5/Scc1.bw
grep -vP 'Mpneumo\t' /data/20220309_Christophe_GC-paper/data/S288c_Mpneumo/ChIP/peaks/CH219/CH219_vs-CH220_genome-S288c_Mpneumo_EHEJFT_peaks.narrowPeak | sed 's,CH219_vs-CH220_genome-S288c_Mpneumo_EHEJFT_,,' > EpigenomicsDataAnalysis_2023/data/day5/Scc1.narrowPeak
cp /data/20220309_Christophe_GC-paper/data/S288c_Mpneumo/MNase/tracks/CH206/CH206^mapped_S288c_Mpneumo^filtered^130-200^HS5L1A.130-200.CPM.bw EpigenomicsDataAnalysis_2023/data/day5/MNase.filtered^130-200bp.bw
cp ~/EpigenomicsDataAnalysis_2023/data/tracks/RNAseq_WT_rep1_R64-1-1.fwd.CPM.bw ~/EpigenomicsDataAnalysis_2023/data/day5/RNAseq.fwd.bw
cp ~/EpigenomicsDataAnalysis_2023/data/tracks/RNAseq_WT_rep1_R64-1-1.rev.CPM.bw ~/EpigenomicsDataAnalysis_2023/data/day5/RNAseq.rev.bw
-->
<section id="inspecting-data-in-igv" class="level2" data-number="21.1">
<h2 data-number="21.1" class="anchored" data-anchor-id="inspecting-data-in-igv"><span class="header-section-number">21.1</span> Inspecting data in IGV</h2>
<ul>
<li>Open <code>.bw</code> files and the Scc1 ChIP-seq peak file in IGV.</li>
<li>Compare the signal of stranded RNA-seq coverage tracks with the location of Scc1 peaks. Comment.</li>
<li>Compare the location of transcription start sites (TSSs) with nucleosome profiles. Comment.</li>
</ul>
</section>
<section id="plotting-stranded-rna-seq-signal-over-scc1-chip-seq-peaks" class="level2" data-number="21.2">
<h2 data-number="21.2" class="anchored" data-anchor-id="plotting-stranded-rna-seq-signal-over-scc1-chip-seq-peaks"><span class="header-section-number">21.2</span> Plotting stranded RNA-seq signal over Scc1 ChIP-seq peaks</h2>
<ul>
<li>Load forward and reverse RNA-seq coverage as <code>Rle</code> in R</li>
<li>Load Scc1 peaks in R</li>
</ul>
<div class="icon callout callout-style-default callout-answer no-icon callout-titled">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-3-contents" aria-controls="callout-3" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Answer
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-3" class="callout-3-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<div class="cell">
<div class="code-with-filename">
<div class="code-with-filename-file">
<pre><strong>R</strong></pre>
</div>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-2"><a href="#cb1-2"></a><span class="fu">library</span>(GenomicRanges)</span>
<span id="cb1-3"><a href="#cb1-3"></a><span class="fu">library</span>(rtracklayer)</span>
<span id="cb1-4"><a href="#cb1-4"></a>rna_fwd_track <span class="ot">&lt;-</span> <span class="fu">import</span>(<span class="st">'data/day5/RNAseq.fwd.bw'</span>, <span class="at">as =</span> <span class="st">'Rle'</span>)</span>
<span id="cb1-5"><a href="#cb1-5"></a>rna_rev_track <span class="ot">&lt;-</span> <span class="fu">import</span>(<span class="st">'data/day5/RNAseq.rev.bw'</span>, <span class="at">as =</span> <span class="st">'Rle'</span>)</span>
<span id="cb1-6"><a href="#cb1-6"></a>scc1_peaks <span class="ot">&lt;-</span> <span class="fu">import</span>(<span class="st">'data/day5/Scc1.narrowPeak'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</div>
</div>
</div>
</div>
<ul>
<li>Resize Scc1 peaks so that they all are centered over their summit (check <code>peak</code> column from the <code>narrowPeak</code> file), and extend then ± 4000 bp</li>
</ul>
<div class="icon callout callout-style-default callout-answer no-icon callout-titled">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-4-contents" aria-controls="callout-4" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Answer
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-4" class="callout-4-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<div class="cell">
<div class="code-with-filename">
<div class="code-with-filename-file">
<pre><strong>R</strong></pre>
</div>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1"></a><span class="fu">library</span>(plyranges)</span>
<span id="cb2-2"><a href="#cb2-2"></a>scc1_peaks <span class="ot">&lt;-</span> scc1_peaks <span class="sc">|&gt;</span> </span>
<span id="cb2-3"><a href="#cb2-3"></a>    <span class="fu">resize</span>(<span class="at">fix =</span> <span class="st">'start'</span>, <span class="at">width =</span> <span class="dv">1</span>) <span class="sc">|&gt;</span> </span>
<span id="cb2-4"><a href="#cb2-4"></a>    <span class="fu">shift_right</span>(scc1_peaks<span class="sc">$</span>peak) <span class="sc">|&gt;</span> </span>
<span id="cb2-5"><a href="#cb2-5"></a>    <span class="fu">resize</span>(<span class="at">fix =</span> <span class="st">'center'</span>, <span class="at">width =</span> <span class="dv">8000</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</div>
</div>
</div>
</div>
<ul>
<li>Compute a <code>seqinfo</code> from the <code>rna_fwd_track</code> <code>Rle</code>, add <code>seqlengths</code> and convert it into a <code>GRanges</code> object.</li>
<li>Remove the extended Scc1 peaks that lie outside of genome boundaries. To do this, check the operators <code>%over%</code> and <code>%within%</code>.</li>
</ul>
<div class="icon callout callout-style-default callout-answer no-icon callout-titled">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-5-contents" aria-controls="callout-5" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Answer
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-5" class="callout-5-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<div class="cell">
<div class="code-with-filename">
<div class="code-with-filename-file">
<pre><strong>R</strong></pre>
</div>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1"></a>genome <span class="ot">&lt;-</span> <span class="fu">seqinfo</span>(rna_fwd_track)</span>
<span id="cb3-2"><a href="#cb3-2"></a><span class="fu">seqlengths</span>(genome) <span class="ot">&lt;-</span> <span class="fu">lengths</span>(rna_fwd_track)</span>
<span id="cb3-3"><a href="#cb3-3"></a>genome <span class="ot">&lt;-</span> <span class="fu">as</span>(genome, <span class="st">"GRanges"</span>)</span>
<span id="cb3-4"><a href="#cb3-4"></a><span class="fu">table</span>(scc1_peaks <span class="sc">%over%</span> genome)</span>
<span id="cb3-5"><a href="#cb3-5"></a><span class="fu">table</span>(scc1_peaks <span class="sc">%within%</span> genome)</span>
<span id="cb3-6"><a href="#cb3-6"></a>scc1_peaks <span class="ot">&lt;-</span> scc1_peaks[scc1_peaks <span class="sc">%within%</span> genome]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</div>
</div>
</div>
</div>
<ul>
<li>Extract the forward RNA-seq coverage signal over the first Scc1 peak</li>
<li>Convert it into a tibble and add coordinates (± 4000 bp centered over Scc1 peak)</li>
<li>Do the same for reverse RNA-seq coverage</li>
<li>Plot the two stranded RNA-seq coverages together</li>
</ul>
<div class="icon callout callout-style-default callout-answer no-icon callout-titled">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-6-contents" aria-controls="callout-6" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Answer
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-6" class="callout-6-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<div class="cell">
<div class="code-with-filename">
<div class="code-with-filename-file">
<pre><strong>R</strong></pre>
</div>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1"></a>rna_fwd_track[scc1_peaks][[<span class="dv">1</span>]]</span>
<span id="cb4-2"><a href="#cb4-2"></a>df <span class="ot">&lt;-</span> rna_fwd_track[scc1_peaks][[<span class="dv">1</span>]] <span class="sc">|&gt;</span> </span>
<span id="cb4-3"><a href="#cb4-3"></a>    <span class="fu">as_tibble</span>() <span class="sc">|&gt;</span></span>
<span id="cb4-4"><a href="#cb4-4"></a>    <span class="fu">mutate</span>(<span class="at">position =</span> <span class="fu">seq</span>(<span class="sc">-</span><span class="dv">3999</span>, <span class="dv">4000</span>, <span class="dv">1</span>))</span>
<span id="cb4-5"><a href="#cb4-5"></a>df<span class="sc">$</span>forward <span class="ot">&lt;-</span> df<span class="sc">$</span>value</span>
<span id="cb4-6"><a href="#cb4-6"></a>df<span class="sc">$</span>reverse <span class="ot">&lt;-</span> rna_rev_track[scc1_peaks][[<span class="dv">1</span>]] <span class="sc">|&gt;</span> <span class="fu">as.vector</span>()</span>
<span id="cb4-7"><a href="#cb4-7"></a>df<span class="sc">$</span>value <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb4-8"><a href="#cb4-8"></a>p <span class="ot">&lt;-</span> <span class="fu">pivot_longer</span>(df, <span class="sc">-</span>pos, <span class="at">names_to =</span> <span class="st">'strandness'</span>, <span class="at">values_to =</span> <span class="st">'coverage'</span>) <span class="sc">|&gt;</span></span>
<span id="cb4-9"><a href="#cb4-9"></a>    <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> pos, <span class="at">y =</span> coverage, <span class="at">col =</span> strandness)) <span class="sc">+</span> </span>
<span id="cb4-10"><a href="#cb4-10"></a>    <span class="fu">geom_line</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</div>
</div>
</div>
</div>
<ul>
<li>Iterate over all Scc1 peaks to extract stranded RNA-seq coverage around Scc1 peaks</li>
</ul>
<div class="icon callout callout-style-default callout-answer no-icon callout-titled">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-7-contents" aria-controls="callout-7" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Answer
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-7" class="callout-7-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<div class="cell">
<div class="code-with-filename">
<div class="code-with-filename-file">
<pre><strong>R</strong></pre>
</div>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1"></a>df <span class="ot">&lt;-</span> rna_fwd_track[scc1_peaks] <span class="sc">|&gt;</span></span>
<span id="cb5-2"><a href="#cb5-2"></a>    <span class="fu">as_tibble</span>() <span class="sc">|&gt;</span> <span class="do">## Convert into a data.frame</span></span>
<span id="cb5-3"><a href="#cb5-3"></a>    <span class="fu">mutate</span>(<span class="at">position =</span> <span class="fu">rep</span>(<span class="fu">seq</span>(<span class="sc">-</span><span class="dv">3999</span>, <span class="dv">4000</span>, <span class="dv">1</span>), <span class="fu">length</span>(scc1_peaks))) <span class="sc">|&gt;</span> <span class="do">## Repeat the position (± 4000bp) as many times as the number of Scc1 peaks</span></span>
<span id="cb5-4"><a href="#cb5-4"></a>    <span class="fu">mutate</span>(<span class="at">forward =</span> value) <span class="sc">|&gt;</span> </span>
<span id="cb5-5"><a href="#cb5-5"></a>    <span class="fu">select</span>(group, position, forward) <span class="sc">|&gt;</span> <span class="do">## Tidy up columns</span></span>
<span id="cb5-6"><a href="#cb5-6"></a>    <span class="fu">mutate</span>(<span class="at">reverse =</span> rna_rev_track[scc1_peaks] <span class="sc">|&gt;</span> <span class="fu">unlist</span>() <span class="sc">|&gt;</span> <span class="fu">as.vector</span>()) <span class="sc">|&gt;</span> <span class="do">## Add reverse scores</span></span>
<span id="cb5-7"><a href="#cb5-7"></a>    <span class="fu">pivot_longer</span>(<span class="sc">-</span><span class="fu">c</span>(group, position), <span class="at">names_to =</span> <span class="st">'strandness'</span>, <span class="at">values_to =</span> <span class="st">'coverage'</span>) <span class="sc">|&gt;</span></span>
<span id="cb5-8"><a href="#cb5-8"></a>    <span class="fu">group_by</span>(position, strandness) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</div>
</div>
</div>
</div>
<ul>
<li>Calculate mean, standard deviation and 95% confidence interval of forward and reverse RNA-seq coverage around Scc1 peaks</li>
<li>Plot the average forward and reverse RNA-seq coverages, with a ribbon showing the 95% confidence interval.</li>
</ul>
<div class="icon callout callout-style-default callout-answer no-icon callout-titled">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-8-contents" aria-controls="callout-8" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Answer
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-8" class="callout-8-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<div class="cell">
<div class="code-with-filename">
<div class="code-with-filename-file">
<pre><strong>R</strong></pre>
</div>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1"></a>aggr_df <span class="ot">&lt;-</span> <span class="fu">summarize</span>(df, </span>
<span id="cb6-2"><a href="#cb6-2"></a>    <span class="at">mean =</span> <span class="fu">mean</span>(coverage), </span>
<span id="cb6-3"><a href="#cb6-3"></a>    <span class="at">sd =</span> <span class="fu">sd</span>(coverage, <span class="at">na.rm =</span> <span class="cn">TRUE</span>), </span>
<span id="cb6-4"><a href="#cb6-4"></a>    <span class="at">n =</span> dplyr<span class="sc">::</span><span class="fu">n</span>()</span>
<span id="cb6-5"><a href="#cb6-5"></a>) <span class="sc">|&gt;</span></span>
<span id="cb6-6"><a href="#cb6-6"></a>    <span class="fu">mutate</span>(</span>
<span id="cb6-7"><a href="#cb6-7"></a>        <span class="at">se =</span> sd <span class="sc">/</span> <span class="fu">sqrt</span>(n),</span>
<span id="cb6-8"><a href="#cb6-8"></a>        <span class="at">CI =</span> <span class="fu">qt</span>(<span class="fl">0.975</span>, n<span class="dv">-1</span>) <span class="sc">*</span> se,</span>
<span id="cb6-9"><a href="#cb6-9"></a>        <span class="at">topMargin =</span> mean <span class="sc">+</span> CI,</span>
<span id="cb6-10"><a href="#cb6-10"></a>        <span class="at">bottomMargin =</span> mean <span class="sc">-</span> CI</span>
<span id="cb6-11"><a href="#cb6-11"></a>    )</span>
<span id="cb6-12"><a href="#cb6-12"></a>p <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(aggr_df, <span class="fu">aes</span>(<span class="at">x =</span> position, <span class="at">y =</span> mean, <span class="at">col =</span> strandness, <span class="at">fill =</span> strandness)) <span class="sc">+</span> </span>
<span id="cb6-13"><a href="#cb6-13"></a>    <span class="fu">geom_line</span>() <span class="sc">+</span> </span>
<span id="cb6-14"><a href="#cb6-14"></a>    <span class="fu">geom_ribbon</span>(<span class="fu">aes</span>(<span class="at">ymax =</span> mean <span class="sc">+</span> CI, <span class="at">ymin =</span> mean <span class="sc">-</span> CI), <span class="at">alpha =</span> <span class="fl">0.2</span>, <span class="at">col =</span> <span class="cn">NA</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</div>
</div>
</div>
</div>
</section>
<section id="plot-a-heaetmap-of-rna-seq-coverage-over-ordered-scc1-peaks" class="level2" data-number="21.3">
<h2 data-number="21.3" class="anchored" data-anchor-id="plot-a-heaetmap-of-rna-seq-coverage-over-ordered-scc1-peaks"><span class="header-section-number">21.3</span> Plot a heaetmap of RNA-seq coverage over ordered Scc1 peaks</h2>
<ul>
<li>Order peaks according to their Scc1 peak signal</li>
<li>Re-compute the stranded RNA-seq coverages around <strong>ordered</strong> Scc1 peaks</li>
<li>Plot a (rasterized) heatmap using <code>geom_tile()</code>, with the distance from Scc1 peak in abscisse and each row representing an individual Scc1 peak locus. You can split forward and reverse coverage with <code>facet_wrap(~ strandness)</code>.</li>
</ul>
<div class="icon callout callout-style-default callout-answer no-icon callout-titled">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-9-contents" aria-controls="callout-9" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Answer
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-9" class="callout-9-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<div class="cell">
<div class="code-with-filename">
<div class="code-with-filename-file">
<pre><strong>R</strong></pre>
</div>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1"></a>ordered_scc1_peaks <span class="ot">&lt;-</span> scc1_peaks[<span class="fu">order</span>(scc1_peaks<span class="sc">$</span>signalValue, <span class="at">decreasing =</span> <span class="cn">TRUE</span>)]</span>
<span id="cb7-2"><a href="#cb7-2"></a>df <span class="ot">&lt;-</span> rna_fwd_track[ordered_scc1_peaks] <span class="sc">|&gt;</span></span>
<span id="cb7-3"><a href="#cb7-3"></a>    <span class="fu">as_tibble</span>() <span class="sc">|&gt;</span> <span class="do">## Convert into a data.frame</span></span>
<span id="cb7-4"><a href="#cb7-4"></a>    <span class="fu">mutate</span>(<span class="at">position =</span> <span class="fu">rep</span>(<span class="fu">seq</span>(<span class="sc">-</span><span class="dv">3999</span>, <span class="dv">4000</span>, <span class="dv">1</span>), <span class="fu">length</span>(ordered_scc1_peaks))) <span class="sc">|&gt;</span> <span class="do">## Repeat the position (± 4000bp) as many times as the number of Scc1 peaks</span></span>
<span id="cb7-5"><a href="#cb7-5"></a>    <span class="fu">mutate</span>(<span class="at">forward =</span> value) <span class="sc">|&gt;</span> </span>
<span id="cb7-6"><a href="#cb7-6"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(group, position, forward) <span class="sc">|&gt;</span> <span class="do">## Tidy up columns</span></span>
<span id="cb7-7"><a href="#cb7-7"></a>    <span class="fu">mutate</span>(<span class="at">reverse =</span> rna_rev_track[ordered_scc1_peaks] <span class="sc">|&gt;</span> <span class="fu">unlist</span>() <span class="sc">|&gt;</span> <span class="fu">as.vector</span>()) <span class="sc">|&gt;</span> <span class="do">## Add reverse scores</span></span>
<span id="cb7-8"><a href="#cb7-8"></a>    <span class="fu">pivot_longer</span>(<span class="sc">-</span><span class="fu">c</span>(group, position), <span class="at">names_to =</span> <span class="st">'strandness'</span>, <span class="at">values_to =</span> <span class="st">'coverage'</span>) <span class="sc">|&gt;</span></span>
<span id="cb7-9"><a href="#cb7-9"></a>    <span class="fu">group_by</span>(position, strandness) </span>
<span id="cb7-10"><a href="#cb7-10"></a>p <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(df, <span class="fu">aes</span>(<span class="at">x =</span> position, <span class="at">y =</span> group, <span class="at">fill =</span> <span class="fu">log2</span>(coverage))) <span class="sc">+</span> </span>
<span id="cb7-11"><a href="#cb7-11"></a>    <span class="fu">geom_tile</span>() <span class="sc">|&gt;</span> ggrastr<span class="sc">::</span><span class="fu">rasterise</span>() <span class="sc">+</span> </span>
<span id="cb7-12"><a href="#cb7-12"></a>    <span class="fu">facet_wrap</span>(<span class="sc">~</span> strandness, <span class="at">scales =</span> <span class="st">'free'</span>) <span class="sc">+</span></span>
<span id="cb7-13"><a href="#cb7-13"></a>    <span class="fu">scale_y_reverse</span>() <span class="sc">+</span></span>
<span id="cb7-14"><a href="#cb7-14"></a>    <span class="fu">scale_fill_gradientn</span>(</span>
<span id="cb7-15"><a href="#cb7-15"></a>        <span class="at">colours =</span> <span class="fu">c</span>(<span class="st">"#FFFEF9"</span>, <span class="st">"#FFCA67"</span>, <span class="st">"#ED3024"</span>, <span class="st">"#1A0A10"</span>), </span>
<span id="cb7-16"><a href="#cb7-16"></a>        <span class="at">na.value =</span> <span class="st">'white'</span></span>
<span id="cb7-17"><a href="#cb7-17"></a>    )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</div>
</div>
</div>
</div>
</section>
<section id="plot-stranded-rna-seq-coverages-for-scc1-peaks-grouped-by-their-strength" class="level2" data-number="21.4">
<h2 data-number="21.4" class="anchored" data-anchor-id="plot-stranded-rna-seq-coverages-for-scc1-peaks-grouped-by-their-strength"><span class="header-section-number">21.4</span> Plot stranded RNA-seq coverages for Scc1 peaks grouped by their strength</h2>
<ul>
<li>Group the coverage tibble in 4 groups, containing the 0-25% (weakest) peaks, then 25-50%, 50-75% and 75-100% (strongest) peaks. This can be done with the <code>ntile</code> function.</li>
<li>Re-plot the average forward and reverse RNA-seq coverages, with a ribbon showing the 95% confidence interval, <strong>splitting the peaks from each group in a different facet (with <code>facet_wrap(~ group)</code>)</strong>.</li>
</ul>
<div class="icon callout callout-style-default callout-answer no-icon callout-titled">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-10-contents" aria-controls="callout-10" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Answer
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-10" class="callout-10-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<div class="cell">
<div class="code-with-filename">
<div class="code-with-filename-file">
<pre><strong>R</strong></pre>
</div>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1"></a>df<span class="sc">$</span>signalValue <span class="ot">&lt;-</span> ordered_scc1_peaks<span class="sc">$</span>signalValue[df<span class="sc">$</span>group]</span>
<span id="cb8-2"><a href="#cb8-2"></a>df<span class="sc">$</span>group <span class="ot">&lt;-</span> <span class="fu">ntile</span>(df<span class="sc">$</span>signalValue, <span class="dv">4</span>)</span>
<span id="cb8-3"><a href="#cb8-3"></a>aggr_df <span class="ot">&lt;-</span> <span class="fu">group_by</span>(df, group, position, strandness) <span class="sc">|&gt;</span></span>
<span id="cb8-4"><a href="#cb8-4"></a>    <span class="fu">summarize</span>( </span>
<span id="cb8-5"><a href="#cb8-5"></a>        <span class="at">mean =</span> <span class="fu">mean</span>(coverage), </span>
<span id="cb8-6"><a href="#cb8-6"></a>        <span class="at">sd =</span> <span class="fu">sd</span>(coverage, <span class="at">na.rm =</span> <span class="cn">TRUE</span>), </span>
<span id="cb8-7"><a href="#cb8-7"></a>        <span class="at">n =</span> dplyr<span class="sc">::</span><span class="fu">n</span>()</span>
<span id="cb8-8"><a href="#cb8-8"></a>    ) <span class="sc">|&gt;</span></span>
<span id="cb8-9"><a href="#cb8-9"></a>    <span class="fu">mutate</span>(</span>
<span id="cb8-10"><a href="#cb8-10"></a>        <span class="at">se =</span> sd <span class="sc">/</span> <span class="fu">sqrt</span>(n),</span>
<span id="cb8-11"><a href="#cb8-11"></a>        <span class="at">CI =</span> <span class="fu">qt</span>(<span class="fl">0.975</span>, n<span class="dv">-1</span>) <span class="sc">*</span> se,</span>
<span id="cb8-12"><a href="#cb8-12"></a>        <span class="at">topMargin =</span> mean <span class="sc">+</span> CI,</span>
<span id="cb8-13"><a href="#cb8-13"></a>        <span class="at">bottomMargin =</span> mean <span class="sc">-</span> CI</span>
<span id="cb8-14"><a href="#cb8-14"></a>    )</span>
<span id="cb8-15"><a href="#cb8-15"></a>p <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(aggr_df, <span class="fu">aes</span>(<span class="at">x =</span> position, <span class="at">y =</span> mean, <span class="at">col =</span> strandness, <span class="at">fill =</span> strandness)) <span class="sc">+</span> </span>
<span id="cb8-16"><a href="#cb8-16"></a>    <span class="fu">geom_line</span>() <span class="sc">+</span> </span>
<span id="cb8-17"><a href="#cb8-17"></a>    <span class="fu">geom_ribbon</span>(<span class="fu">aes</span>(<span class="at">ymax =</span> mean <span class="sc">+</span> CI, <span class="at">ymin =</span> mean <span class="sc">-</span> CI), <span class="at">alpha =</span> <span class="fl">0.2</span>, <span class="at">col =</span> <span class="cn">NA</span>) <span class="sc">+</span></span>
<span id="cb8-18"><a href="#cb8-18"></a>    <span class="fu">facet_wrap</span>(<span class="sc">~</span> group)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</div>
</div>
</div>
</div>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="../../content/day5/Demo5_Bioc-resources.html" class="pagination-link">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">20</span>&nbsp; <span class="chapter-title">Demo 5 - Leveraging Bioconductor resources</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="../../content/extra.html" class="pagination-link">
        <span class="nav-page-text">Extra resources</span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-left">Epigenomics Data Analysis |<br>
J. Serizay</div>   
    <div class="nav-footer-center">
      &nbsp;
    </div>
    <div class="nav-footer-right">This book was built with <a href="https://quarto.org/">Quarto</a>.</div>
  </div>
</footer>



</body></html>